lib/ansible/playbook/role/requirement.py

```diff
@@ -105,9 +105,7 @@ def role_yaml_parse(role):
                     role["src"] = "git+" + role["src"]
 
                 if '+' in role["src"]:
-                    (scm, src) = role["src"].split('+')
-                    role["scm"] = scm
-                    role["src"] = src
+                    role["scm"], dummy, role["src"] = role["src"].partition('+')
```

Comment:
It looks like `ansible.module_utils.urls.generic_urlparse()` may be useful for parsing the URL here.

Reply:
With `generic_urlparse`, there is additional data manipulation is required 

```
generic_urlparse(urlparse(role['src']))
{'scheme': 'git+https', 'netloc': 'gitlab+deploy-token-312644:_aJQ9c3HWzmRR4knBNyx@gitlab.com', 'path': '/akasurde/ansible-demo', 'params': '', 'query': '', 'fragment': '', 'username': 'gitlab+deploy-token-312644', 'password': '_aJQ9c3HWzmRR4knBNyx', 'hostname': 'gitlab.com', 'port': None}
```

I would like to keep it simple for now.

Reply:
Ah, ok. It was worth looking at.